flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/translators/SinkTransformationTranslator.java

```diff
@@ -0,0 +1,283 @@
+				Sink.class,
+				sinkTransformation.getSink().getClass(),
+				1,
+				null,
+				null);
+
+		final int parallelism = getParallelism(sinkTransformation, context);
+
+		return addOperatorToStreamGraph(
+				committerFactory, inputId,
+				committableTypeInfo,
+				committableTypeInfo,
+				"Sink Committer:",
+				parallelism,
+				sinkTransformation.getMaxParallelism(),
+				sinkTransformation,
+				context);
+	}
+
+	/**
+	 * Try to add a sink global committer to the stream graph.
+	 *
+	 * @param inputId The global committer's input stream node id.
+	 * @param sinkTransformation The transformation that the global committer belongs to
+	 * @param globalCommitterFactory The global committer factory
+	 */
+	private void addGlobalCommitter(
+			int inputId,
+			SinkTransformation<InputT, CommT, WriterStateT, GlobalCommT> sinkTransformation,
+			OneInputStreamOperatorFactory<CommT, GlobalCommT> globalCommitterFactory,
+			Context context) {
+
+		if (!sinkTransformation.getSink().getGlobalCommittableSerializer().isPresent()) {
+			return;
+		}
+
+		final TypeInformation<CommT> committableTypeInfo = TypeExtractor.createTypeInfo(
+				Sink.class,
+				sinkTransformation.getSink().getClass(),
+				1,
+				null,
+				null);
+
+		addOperatorToStreamGraph(
+				globalCommitterFactory, inputId,
+				committableTypeInfo,
+				null,
+				"Sink Global Committer:",
+				1,
+				1,
+				sinkTransformation,
+				context);
+	}
+
+	private int getParallelism(
+			SinkTransformation<InputT, CommT, WriterStateT, GlobalCommT> sinkTransformation,
+			Context context) {
+		return sinkTransformation.getParallelism() != ExecutionConfig.PARALLELISM_DEFAULT
+				? sinkTransformation.getParallelism()
+				: context.getStreamGraph().getExecutionConfig().getParallelism();
+	}
+
+	/**
+	 * Add a operator to the {@link StreamGraph}.
+	 *
+	 * @param operatorFactory The operator factory
+	 * @param inputId The upstream stream node id of the operator
+	 * @param inTypeInfo The input type information of the operator
+	 * @param outTypInfo The output type information of the operator
+	 * @param prefix The prefix of the name and uid of the operator
+	 * @param parallelism The parallelism of the operator
+	 * @param maxParallelism The max parallelism of the operator
+	 * @param sinkTransformation The sink transformation which the operator belongs to
+	 *
+	 * @return The stream node id of the operator
+	 */
+	private <IN, OUT> int addOperatorToStreamGraph(
+			StreamOperatorFactory<OUT> operatorFactory, int inputId,
+			TypeInformation<IN> inTypeInfo,
+			TypeInformation<OUT> outTypInfo,
+			String prefix,
+			int parallelism,
+			int maxParallelism,
+			SinkTransformation<InputT, CommT, WriterStateT, GlobalCommT> sinkTransformation,
+			Context context) {
+		final StreamGraph streamGraph = context.getStreamGraph();
+		final String slotSharingGroup = context.getSlotSharingGroup();
+		final int transformationId = Transformation.getNewNodeId();
+
```

Comment:
Do we want this setting to apply it to all operators or only the `Writer`? I think that it seems more relevant for the writer only as that is what the user "sees", but I am not 100% sure.

Reply:
Do you mean whether we want to set the same slot sharing group to all these three operators or not? 

I think the answer might be yes. It is because this would let the operator use less slots. Specifically:
In case of the streaming execution mode, the same share slot group could make `Writer` and `Committer` be chained together and make the `GlobalComitter` reuse the same slot.
In case of the batch execution mode, the same slot group could make `Committer` and `GlobalCommitter` be chained together if the parallelism of `Committer` is 1.

Of course maybe I miss something. What do you think?

Reply:
Sorry, I was referring to the `chainingStrategy`.

Reply:
I think you are right. It is more intuitive to only set to the `writer`